### PR TITLE
refactor: Rewrite the ESLint config file for this repo with Flat Config

### PR DESCRIPTION
**Pull request by wakamsha**
## What changed / motivation ?

When ESLint is updated to v9 in #354, `eslintrc` will be deprecated, and it will become difficult to use `.eslintrc.js` for this repository. Therefore, I rewrote `.eslintrc.js` as `eslint.config.mjs` in advance.

In addition, as a result of this, invalid code that was not detected before is now properly detected, so I fixed them at this time.

## Linked PR / Issues

- #354

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Chore (routine task, maintenance, or non-functional change that doesn't modify src or test files)

## Pre-flight checklist

- [x] I have read the [Contributing Guidelines](https://github.com/moneyforward/eslint-config-moneyforward/blob/main/.github/CONTRIBUTING.md)
- [x] Performed a self-review of my code

### CODE REVIEW

File: configs/flat/index.mjs
Comment (wakamsha) [inline configs/flat/index.mjs:1]: fixed with [`unicorn/prefer-export-from`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-export-from.md) rule.

File: rules/imports.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ module.exports = {
     'import/core-modules': [],
     'import/ignore': [
       'node_modules',
-      '\\.(coffee|scss|css|less|hbs|svg|json|jpg|jpeg|png|webp)$',
+      String.raw`\.(coffee|scss|css|less|hbs|svg|json|jpg|jpeg|png|webp)$`,
```
Comment (wakamsha) [inline rules/imports.js:31]: fixed with [`unicorn/prefer-string-raw`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-raw.md)

File: rules/variables.js [hunk 2]
```diff
@@ -39,12 +39,11 @@ module.exports = {
         name: 'isNaN',
         message: 'Use Number.isNaN instead.',
       },
-    ].concat(
-      confusingBrowserGlobals.map((g) => ({
+      ...confusingBrowserGlobals.map((g) => ({
```
Comment (wakamsha) [inline rules/variables.js:42]: fixed with [`unicorn/prefer-spread`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-spread.md) rule.

File: tests/requires.test.js [hunk 3]
```diff
@@ -1,46 +1,42 @@
 /* eslint-disable import/no-dynamic-require */
 /* eslint-disable n/global-require */
-const { readdirSync } = require('fs');
-const { resolve, join } = require('path');
+const { readdirSync } = require('node:fs');
+const path = require('node:path');
```
Comment (wakamsha) [inline tests/requires.test.js:4]: fixed with [`unicorn/prefer-node-protocol`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-node-protocol.md) rule.

File: tests/requires.test.js [hunk 4]
```diff
@@ -1,46 +1,42 @@
...
-  const dir = resolve(join(__dirname, '../rules'));
+  const directory = path.resolve(path.join(__dirname, '../rules'));
 
-  const files = readdirSync(dir, {
+  const files = readdirSync(directory, {
     withFileTypes: true,
-  })
-    .map((dirnet) => extractFilePath(dirnet, dir))
-    .flat();
+  }).flatMap((dirnet) => extractFilePath(dirnet, directory));
```
Comment (wakamsha) [inline tests/requires.test.js:23]: fixed with [`unicorn/prefer-array-flat-map`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-array-flat-map.md) rule.

File: tests/requires.test.js [hunk 5]
```diff
@@ -1,46 +1,42 @@
 /* eslint-disable import/no-dynamic-require */
 /* eslint-disable n/global-require */
-const { readdirSync } = require('fs');
-const { resolve, join } = require('path');
+const { readdirSync } = require('node:fs');
+const path = require('node:path');
 
 test('parsable all eslintrc format configs', () => {
-  const dir = resolve(join(__dirname, '../configs/eslintrc'));
+  const directory = path.resolve(path.join(__dirname, '../configs/eslintrc'));
```
Comment (wakamsha) [inline tests/requires.test.js:7]: fixed with [`unicorn/prevent-abbreviations`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v56.0.1/docs/rules/prevent-abbreviations.md) rule.

File: tests/snapshot-test/eslintrc/node/dummy.js
Comment (wakamsha) [inline tests/snapshot-test/eslintrc/node/dummy.js:1]: This file was installed for snapshot testing, but it was found to be unnecessary so I deleted it.

**Review by taigakiyokawa — APPROVED** [review]
LGTM 🙆‍♂️ Thank you!

**Pull request merged**